go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+		_ = tablet.VttabletProcess.Setup()
+	}
+
+	// rdonly tablets will be started
+	for _, tablet := range []cluster.Vttablet{sourceRdOnlyTablet1, sourceRdOnlyTablet2, destinationRdOnlyTablet1, destinationRdOnlyTablet2} {
+		_ = tablet.VttabletProcess.Setup()
+	}
+
+	// check SrvKeyspace
+	ksServedFrom := "ServedFrom(master): source_keyspace\nServedFrom(rdonly): source_keyspace\nServedFrom(replica): source_keyspace\n"
+	checkSrvKeyspaceServedFrom(t, cellj, destinationKeyspace, ksServedFrom, *clusterInstance)
+
+	// reparent to make the tablets work (we use health check, fix their types)
+	err = clusterInstance.VtctlclientProcess.InitShardMaster(sourceKeyspace, shardName, cellj, sourceMasterTablet.TabletUID)
+	assert.Nil(t, err)
+	err = clusterInstance.VtctlclientProcess.InitShardMaster(destinationKeyspace, shardName, cellj, destinationMasterTablet.TabletUID)
+	assert.Nil(t, err)
+
+	sourceMasterTablet.Type = "master"
+	destinationMasterTablet.Type = "master"
+
+	for _, tablet := range []cluster.Vttablet{sourceReplicaTablet, destinationReplicaTablet} {
+		_ = tablet.VttabletProcess.WaitForTabletType("SERVING")
+		assert.Nil(t, err)
+	}
+	for _, tablet := range []cluster.Vttablet{sourceRdOnlyTablet1, sourceRdOnlyTablet2, destinationRdOnlyTablet1, destinationRdOnlyTablet2} {
+		_ = tablet.VttabletProcess.WaitForTabletType("SERVING")
+		assert.Nil(t, err)
+	}
+
+	for _, tablet := range []cluster.Vttablet{sourceMasterTablet, destinationMasterTablet, sourceReplicaTablet, destinationReplicaTablet, sourceRdOnlyTablet1, sourceRdOnlyTablet2, destinationRdOnlyTablet1, destinationRdOnlyTablet2} {
+		assert.Equal(t, tablet.VttabletProcess.GetTabletStatus(), "SERVING")
+	}
+
+	err = clusterInstance.StartVtgate()
+	assert.Nil(t, err)
+
+	err = clusterInstance.VtgateProcess.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.master", sourceKeyspace, shardName), 1)
+	assert.Nil(t, err)
+	err = clusterInstance.VtgateProcess.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.replica", sourceKeyspace, shardName), 1)
+	assert.Nil(t, err)
+	err = clusterInstance.VtgateProcess.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.rdonly", sourceKeyspace, shardName), 2)
+	assert.Nil(t, err)
+	err = clusterInstance.VtgateProcess.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.master", destinationKeyspace, shardName), 1)
+	assert.Nil(t, err)
+	err = clusterInstance.VtgateProcess.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.replica", destinationKeyspace, shardName), 1)
+	assert.Nil(t, err)
+	err = clusterInstance.VtgateProcess.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.rdonly", destinationKeyspace, shardName), 2)
+	assert.Nil(t, err)
+
+	vtParams := mysql.ConnParams{
+		Host: clusterInstance.Hostname,
+		Port: clusterInstance.VtgateMySQLPort,
+	}
+
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
```

Comment:
Should we move this just after we start vtgate?

Reply:
Done. Refer [PR46](https://github.com/planetscale/vitess/pull/46)